test/core/2_pendleData.spec.ts

```diff
@@ -30,20 +41,191 @@ describe('PendleData', async () => {
     snapshotId = await evm_snapshot();
   });
 
-  it('should be able to setMarketFees', async () => {
-    await data.setMarketFees(10, 101);
-    let swapFee = await data.swapFee();
-    let protocolSwapFee = await data.protocolSwapFee();
-    expect(swapFee).to.be.eq(10);
-    expect(protocolSwapFee).to.be.eq(101);
+  it('Should be able to deploy with correct initial data', async() => {
+    const treasuryInData = await data.treasury();
+    expect(treasuryInData).to.be.eq(treasury.address);
+    const pausingManagerInData = await data.pausingManager();
+    expect(pausingManagerInData).to.be.eq(pausingManager.address)
+    const routerInData = await data.router();
+    expect(routerInData).to.be.eq(fixture.core.router.address);
   });
 
-  it('should be able to get allMarketsLength', async () => {
+  it('Should not be able to construct with zero address as treasury', async() => {
+    await expect(deployContract(alice, PendleData, [fixture.core.govManager.address, consts.ZERO_ADDRESS, fixture.core.pausingManager.address])).to.be.revertedWith("ZERO_ADDRESS");
```

Comment:
You should add (probably its already added tho) "ZERO_MESSAGE" to ErrMsg.ts and import the error messages object in to prevent hardcoded message strings.